packages/web-components/src/components/page-header/page-header-breadcrumb.ts

```diff
@@ -33,20 +33,101 @@ class CDSPageHeaderBreadcrumb extends LitElement {
+   */
+  @property({ attribute: 'content-actions-flush', type: Boolean })
+  contentActionsFlush = false;
+
+  /**
+   * Set to `true` if there are content actions
+   */
+  @state()
+  private _hasContentActions = false;
+
+  /**
+   * Set to `true` if there are page actions
+   */
+  @state()
+  private _hasPageActions = false;
+
+  /**
+   * Handles `slotchange` event for content actions.
+   */
+  protected _handleContentActionsSlotChange({ target }: Event) {
+    this._hasContentActions = Boolean(
+      (target as HTMLSlotElement).assignedNodes().length
+    );
+    this.requestUpdate();
+  }
+
+  /**
+   * Handles `slotchange` event for page actions.
+   */
+  protected _handlePageActionsSlotChange({ target }: Event) {
+    this._hasPageActions = Boolean(
+      (target as HTMLSlotElement).assignedNodes().length
+    );
+    this.requestUpdate();
+  }
+
+  render() {
+    const {
+      border,
+      withinGrid,
+      pageActionsFlush,
+      contentActionsFlush,
+      _hasContentActions: hasContentActions,
+      _hasPageActions: hasPageActions,
+      _handleContentActionsSlotChange: handleContentActionsSlotChange,
+      _handlePageActionsSlotChange: handlePageActionsSlotChange,
+    } = this;
+    console.log(border);
     const gridClasses = classMap({
-      [`${prefix}--grid`]: !withinGrid,
+      [`${prefix}--css-grid`]: !withinGrid,
       [`${prefix}--subgrid ${prefix}--subgrid--wide`]: withinGrid,
     });
 
+    const breadcrumbBarClasses = classMap({
+      [`${prefix}--page-header__breadcrumb-bar`]: true,
+      [`${prefix}--page-header__breadcrumb-bar-border`]: border,
+      [`${prefix}--page-header__breadcrumb__actions-flush`]: pageActionsFlush,
+    });
+
+    const contentActionsClasses = classMap({
+      [`${prefix}--page-header__breadcrumb__content-actions`]:
+        !contentActionsFlush,
+    });
+
     return html`
-      <div class="${gridClasses}">
-        <div
-          class="${prefix}--sm:col-span-4 ${prefix}--md:col-span-8 ${prefix}--lg:col-span-16 ${prefix}--css-grid-column">
-          <div class="${prefix}--page-header__breadcrumb-container">
-            <slot></slot>
+      <div class="${breadcrumbBarClasses}">
+        <div class="${gridClasses}">
+          <div
+            class="${prefix}--sm:col-span-4 ${prefix}--md:col-span-8 ${prefix}--lg:col-span-16 ${prefix}--css-grid-column">
+            <div class="${prefix}--page-header__breadcrumb-container">
+              <div class="${prefix}--page-header__breadcrumb-wrapper">
+                <slot name="icon"></slot>
+                <slot></slot>
+              </div>
+              <div class="${prefix}--page-header__breadcrumb__actions">
+                <div class="${contentActionsClasses}">
+                  <slot
+                    name="content-actions"
+                    @slotchange=${handleContentActionsSlotChange}>
+                  </slot>
+                </div>
```

Comment:
```suggestion
                  <slot
                    name="content-actions"
                    @slotchange=${handleContentActionsSlotChange}>
                  </slot>
```

From my comment above, you might be able to remove the wrapper div?